samza-core/src/test/java/org/apache/samza/zk/TestZkUtils.java

```diff
@@ -89,11 +91,11 @@ public static void teardown() {
 
   @Test
   public void testRegisterProcessorId() {
-    String assignedPath = zkUtils.registerProcessorAndGetId("0.0.0.0");
+    String assignedPath = zkUtils.registerProcessorAndGetId("0.0.0.0 1");
```

Comment:
Looks like there is some specific data format expected by this method - registerProcessorAndGetId. maybe it will be cleaner to change the API to take two arguments, as in registerProcessorAndGetId(String hostname, String processorId). Serialization format can be completely hidden from the caller. Same thing applies when reading the data from the ZK. Deserialization should be hidden from the caller if ZkUtils. 

As a side question: Is this "hostname" used anywhere?